app/decorators/physical_switch_decorator.rb

```diff
@@ -0,0 +1,43 @@
+class PhysicalSwitchDecorator < MiqDecorator
+  def self.fonticon
+    'ff ff-network-switch'
+  end
+
+  def quadicon
+    {
+      :top_left     => {
+        :text    => hardware.physical_ports.try(:size).to_i,
+        :tooltip => _('Ports')
+      },
+      :top_right    => {
+        :tooltip => power_state.try(:downcase),
+      }.merge(QuadiconHelper.machine_state(power_state)),
+      :bottom_left  => {
+        :fonticon => fonticon,
+        :tooltip  => _(type),
```

Comment:
Umm, this is I think a little bit too much, you need the gettext calls only around the strings. But maybe @mzazrivec or @martinpovolny disagrees, so asking them here...

Reply:
@skateman These strings are translated in the list, so I thought that was a good idea to put the translation here too. But I think the translations are not working in the quadicons code. I was about to investigate that, but since it is another issue, I did not investigate yet.

@mzazrivec type is `ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch`.

If you decide to keep the translation here I will update the other Physical Infra pages in another PR.

Reply:
I think instead of `_(type)` you want to use `ui_lookup(:model => type)` here.

Reply:
@mzazrivec @skateman Yes, that works. Can I update the code? If yes I will also create another PR for the others physical infra elements. And about the translation at health_state in this same code?